src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/HtmlHelperOfT.cs

```diff
@@ -96,10 +99,13 @@ public override void Contextualize(ViewContext viewContext)
                 throw new ArgumentNullException(nameof(expression));
             }
 
-            var modelExplorer = ExpressionMetadataProvider.FromLambdaExpression(expression, ViewData, MetadataProvider);
-
-            return GenerateDropDown(modelExplorer, ExpressionHelper.GetExpressionText(expression), selectList,
```

Comment:
Methods in this class were fairly random about `ExpressionMetadataProvider.FromLambdaExpression()` vs `GetModelExplorer()` and `ExpressionHelper.GetExpressionText()` vs `GetExpressionName()`. Change to use the helpers almost everywhere did not affect any tests though the helpers have a couple of additional error cases.